app/src/lib/file-system.ts

```diff
@@ -0,0 +1,20 @@
+import * as fileSystem from 'fs-extra'
+import ErrnoException = NodeJS.ErrnoException
+
+interface IErrorCallbackType { (err: ErrnoException): void}
+
+
+/** this function wraps the filesystem mkdir function and silently returns if
+ * the directory already exists. If a different error was thrown, it's passed to an error handling callback
+ *
+ * @param directoryPath the path of the directory the user wants to create
+ * @param errorCallback function to be called in the event of an unexpected error
+ */
+export async function mkdirIfNeeded (directoryPath: string, errorCallback: IErrorCallbackType) {
```

Comment:
Actually, rather than having an error callback here could we make this more Promise-y? Rough example follows, might not compile:

```ts
export function mkdirIfNeeded(directoryPath: string): Promise<void> {
  return new Promise((resolve, reject) => {
    fs.mkdir(directoryPath, err => {
      if (err && err.code !== 'EEXIST') {
        reject(err)
        return
      }
      resolve()
   })
}
```

This means our `log.ts` example will look like this:

```ts
mkdirIfNeeded(logPath)
   .then(() => resolve(initializeWinston(getLogFilePath(logPath))))
   .catch(error => reject(error))
```

And our new example will look like:

```ts
mkdirIfNeeded(logPath)
  .then(() => shell.showItemInFolder(logPath))
  .catch(err => log('error', err.message))
```